### ISSUE TITLE
presentShortcut main thread warning

### ISSUE THREAD

**Issue by swallville** [Gustash/react-native-siri-shortcut#32]
During the very first execution in devices, function presentShortcut of RNSiriShortcuts.swift gives a runtime warning that modalPresentationStyle could only be used inside the main thread. This causes the Modal presentation being stuck, sometimes for more than 15 seconds, looking like for the average user that the app crashed. I tweaked a bit the function and now is fixed with the if else statement inside the scope of DispatchQueue.main.async:

```
    @available(iOS 12.0, *)
    @objc func presentShortcut(_ jsonOptions: Dictionary<String, Any>, callback: @escaping RCTResponseSenderBlock) {
        presentShortcutCallback = callback
        let activity = ShortcutsModule.generateUserActivity(jsonOptions)
        
        let shortcut = INShortcut(userActivity: activity)
        
        // To preserve compatilibility with iOS >9.0, the array contains NSObjects, so we need to convert here
        let addedVoiceShortcut = (voiceShortcuts as! Array<INVoiceShortcut>).first { (voiceShortcut) -> Bool in
            if let userActivity = voiceShortcut.shortcut.userActivity, userActivity.activityType == activity.activityType {
                return true
            }
            return false
        }

        DispatchQueue.main.async {
            // The shortcut was not added yet, so present a form to add it
            if (addedVoiceShortcut == nil) {
                self.presenterViewController = INUIAddVoiceShortcutViewController(shortcut: shortcut)
                self.presenterViewController!.modalPresentationStyle = .formSheet
                (self.presenterViewController as! INUIAddVoiceShortcutViewController).delegate = self
            } // The shortcut was already added, so we present a form to edit it
            else {
                self.presenterViewController = INUIEditVoiceShortcutViewController(voiceShortcut: addedVoiceShortcut!)
                self.presenterViewController!.modalPresentationStyle = .formSheet
                (self.presenterViewController as! INUIEditVoiceShortcutViewController).delegate = self
            }

            UIApplication.shared.keyWindow!.rootViewController!.present(self.presenterViewController!, animated: true, completion: nil)
        }
    }
```

Could it be fixed in the next release? I tested this fix with an iPhone 6s and an iPhone 11 Pro Max.
Thanks for the amazing job with the lib @Gustash .

**Comment by Gustash** [comment]
I'll take a look at this today and release an update on npm. Thanks for the shout! I find it weird that XCode doesn't warn you about this on compile time...

**Comment by Gustash** [comment]
@swallville this has been fixed on v1.6.4. You can get it from NPM now. Please re-open this issue if this ever occurs again.

**Comment by swallville** [comment]
Thanks for the work @Gustash , really appreciate it.
